src/Network/Network/AzureFirewall/NatRule/NewAzureFirewallNatRuleCommand.cs

```diff
@@ -90,6 +94,12 @@ public override void Execute()
             // Add some validation based on the type of RuleCollection (SNAT will be supported later)
             // if (MNM.AzureFirewallNatRCActionType.Dnat.Equals(ActionType))
             {
+               // One of SourceAddress or SourceIpGroup must be present
+               if ((SourceAddress == null) && (SourceIpGroup == null))
+               {
+                  throw new ArgumentException("Either SourceAddress or SourceIpGroup is required.");
```

Comment:
better to put exception message here https://github.com/Azure/azure-powershell/blob/ipgroup_azfw_branch/src/Network/Network/Properties/Resources.resx